### PR TITLE
T27 - STACK<T>

### PR DESCRIPTION
**Pull request by Ajay07Metamation**

### CODE REVIEW

**Comment by Ajay07Metamation** [comment]
All changes done in the latest commit

File: Training/Program.cs [hunk 1]
```diff
@@ -1,7 +1,79 @@
...
+WriteLine (stack.IsEmpty);
+WriteLine (stack.Pop ());
+stack.Display ();
+WriteLine (stack.Peek ());
+stack.Display ();
+
+/// <summary>Represents a variable size last in first out collection of instances of same specified type</summary>
+class MyStack<T> {
+   // Private variable for creating array and for getting count of elements
+   private T[] mArrayStack;
```
Comment (priyarathna) [inline Training/Program.cs:27]: No need of 'private' explicitly. And private fields should be at the end of the program.

File: Training/Program.cs [hunk 2]
```diff
@@ -1,7 +1,79 @@
-﻿namespace Training {
-    internal class Program {
-        static void Main (string[] args) {
-            Console.WriteLine ("Hello, World!");
-        }
-    }
+﻿// MyStack<T>
+// To create a Stack using array as underlying structure
```
Comment (priyarathna) [inline Training/Program.cs:2]: This comments is not understandable.

File: Training/Program.cs [hunk 3]
```diff
@@ -1,7 +1,79 @@
...
+   public T Peek () {
+      if (IsEmpty) throw new InvalidOperationException ();
+      Write ("Peek element : ");
+      return mArrayStack[mCount - 1];
+   }
+
+   /// <summary>Print the stack</summary>
+   public void Display () {
+      Write ("Stack : ");
+      for (int i = 0; i < mCount; i++) { Write (mArrayStack[i] + " "); }
```
Comment (priyarathna) [inline Training/Program.cs:73]: Flower-brace is not needed for single statements.

File: Training/Program.cs [hunk 4]
```diff
@@ -1,7 +1,79 @@
...
+   }
+   /// <summary>Pop the last inserted element in the stack</summary>
+   /// <returns>Returns the popped Element</returns>
+   /// <exception cref="InvalidOperationException"></exception>
+   public T Pop () {
+      if (IsEmpty) throw new InvalidOperationException ();
+      var poppedItem = mArrayStack[mCount - 1];
+      Array.Clear (mArrayStack, mCount - 1, 1);
+      mCount--;
+      Write ("Popped element : ");
```
Comment (priyarathna) [inline Training/Program.cs:57]: Remove Write/WriteLine in Pop/Peek. Use 'Display' for displaying the stack.

**Review by priyarathna — CHANGES_REQUESTED** [review]

File: Training/Program.cs [hunk 5]
```diff
@@ -1,7 +1,82 @@
-﻿namespace Training {
-    internal class Program {
-        static void Main (string[] args) {
-            Console.WriteLine ("Hello, World!");
-        }
-    }
+﻿// MyStack<T>
+// Implementation of stack using array
+// Test case
```
Comment (priyarathna) [inline Training/Program.cs:3]: Comment is not relevant.

File: Training/Program.cs [hunk 6]
```diff
@@ -1,7 +1,82 @@
...
+stack.Push ('g');
+stack.Push ('h');
+stack.Display ();
+WriteLine (stack.IsEmpty);
+WriteLine ("The popped element is : " + stack.Pop ());
+stack.Display ();
+WriteLine ("The peek element is : " + stack.Peek ());
+stack.Display ();
+
+/// <summary>Represents a variable size last in first out collection of instances of same specified type</summary>
```
Comment (priyarathna) [inline Training/Program.cs:28]: Comment is not relevant.

File: Training/Program.cs [hunk 7]
```diff
@@ -1,7 +1,82 @@
...
+   /// <exception cref="InvalidOperationException"></exception>
+   public T Pop () {
+      if (IsEmpty) throw new InvalidOperationException ();
+      var poppedItem = mArrayStack[mCount - 1];
+      Array.Clear (mArrayStack, mCount - 1, 1);
+      mCount--;
+      return poppedItem;
+   }
+
+   /// <summary>Print the last element inserted at the top of the stack</summary>
```
Comment (priyarathna) [inline Training/Program.cs:61]: Comment is not relevant.

**Review by priyarathna — CHANGES_REQUESTED** [review]

File: Training/Program.cs [hunk 8]
```diff
@@ -1,52 +1,74 @@
...
 
+   #region Constructor -------------------------------------------
    // Constructor initialising the array and count variable
    public MyStack () {
       mArrayStack = new T[4];
       mCount = 0;
    }
+   #endregion
 
+   #region Property --------------------------------------------
```
Comment (priyarathna) [inline Training/Program.cs:52]: Not 'Property'. Change it to 'Properties'.

**Review by priyarathna — CHANGES_REQUESTED** [review]

**Review by priyarathna — APPROVED** [review]

**Pull request closed without merging**